pkg/cmd/server/kubernetes/node.go

```diff
@@ -46,8 +50,14 @@ type NodeConfig struct {
 	// The image used as the Kubelet network namespace and volume container.
 	NetworkContainerImage string
 
-	// A client to connect to etcd
-	EtcdClient *etcd.Client
+	// Whether to enable TLS serving
+	TLS bool
+
+	KubeletCertFile string
+	KubeletKeyFile  string
+
+	// A client to connect to the master.
+	Client *client.Client
```

Comment:
Does the kubelet hand this client off to child pods or controllers? We probably want to nip that in the bud, or at least stub out helper methods so we can distribute clients with more scoped permissions later